ui/sv_icons.py

```diff
@@ -0,0 +1,54 @@
+import bpy
+import os
+import glob
+import bpy.utils.previews
+
+# custom icons dictionary
+_icon_collection = {}
+
+
+def custom_icon(name):
+    load_custom_icons()  # load in case they custom icons not already loaded
+
+    custom_icons = _icon_collection["main"]
+
+    default = lambda: None  # for no icon with given name will return zero
+    default.icon_id = 0
+
+    return custom_icons.get(name, default).icon_id
+
+
+def load_custom_icons():
+    if len(_icon_collection):  # return if custom icons already loaded
```

Comment:
All standard types behave this way. 
Not numpy arrays however... Don't know what they where thinking there. 😢